util/stringutil/string_util.go

```diff
@@ -234,3 +235,8 @@ func DoMatch(str string, patChars, patTypes []byte) bool {
 	}
 	return sIdx == len(str)
 }
+
+// Copy deep copies a string.
```

Comment:
add comment why we need to copy a string. because the string is `hack` from byte(chunk)

Reply:
no need to add it here, should add the comments in the caller side.